### PR TITLE
Missing mocking of http client service calls

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: apps/files_sharing/lib/External/Storage.php [hunk 1]
```diff
@@ -64,10 +64,14 @@ class Storage extends DAV implements ISharedStorage {
...
-		$discoveryManager = new DiscoveryManager(
-			$this->memcacheFactory,
-			\OC::$server->getHTTPClientService()
-		);
+		if (isset($options['discoveryManager'])) {
+			$discoveryManager = $options['discoveryManager'];
+		} else {
+			$discoveryManager = new DiscoveryManager(
+				$this->memcacheFactory,
+				\OC::$server->getHTTPClientService()
```
Comment (PVince81) [inline apps/files_sharing/lib/External/Storage.php]: why not also inject this one as option ? Would prevent some testing gymnastics like https://github.com/owncloud/core/pull/25706/files#diff-32d0c957d19da1b027823e9883ba2cc6R129

**Comment by PVince81** [comment]
Code looks good, see my one comment.
Also let's see if tests will pass...

File: apps/files_sharing/lib/External/Storage.php (hunk 1 above)
Comment (DeepDiver1975) [inline apps/files_sharing/lib/External/Storage.php]: Makes sense - yes - will do ....

**Comment by DeepDiver1975** [comment]
@PVince81 @butonic let's get this in - needed to get the new jenkins setup up and running .... THX

**Comment by butonic** [comment]
👍

**Comment by DeepDiver1975** [comment]
superseeded by https://github.com/owncloud/core/pull/25769

**Pull request closed without merging**